table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Checks if all partitions of a table should be bypassed.
+   *
+   * @param tableName the table name
+   * @return true if the table is configured to be fully bypassed, false otherwise
+   * @see UdbFilterSpec#isBypassedTable(String)
+   */
+  public boolean isFullyBypassedTable(String tableName) {
+    boolean isFullyBypassed = matchedByTemplates(mBypassedTables, tableName);
+    return shadowedByIgnore(tableName, handleInclusionExclusion(mTableBypassMode, isFullyBypassed));
+  }
+
+  /**
+   * Checks if a table is ignored.
+   *
+   * @param tableName the table name
+   * @return true if the table is ignored, false otherwise
+   */
+  public boolean isIgnoredTable(String tableName) {
+    boolean isIgnored = matchedByTemplates(mIgnoredTables, tableName);
+    return handleInclusionExclusion(mIgnoreMode, isIgnored);
+  }
+
+  /**
+   * Checks by a partition's name if it should be bypassed.
+   *
+   * @param tableName the table name
+   * @param partitionName the partition name
+   * @return true if the partition should be bypassed, false otherwise
+   */
+  public boolean isBypassedPartition(String tableName, String partitionName) {
+    if (isFullyBypassedTable(tableName)) {
+      return true;
+    }
+    if (!mBypassedPartitions.containsKey(tableName)) {
+      return false;
+    }
+    Mode partitionMode = mBypassedPartitions.get(tableName).getFirst();
+    boolean isBypassed =
+        matchedByTemplates(mBypassedPartitions.get(tableName).getSecond(), partitionName);
+    return shadowedByIgnore(tableName, handleInclusionExclusion(partitionMode, isBypassed));
+  }
+
+  private static boolean handleInclusionExclusion(Mode mode, boolean selected) {
+    switch (mode) {
+      case INCLUDE:
+        return selected;
+      case EXCLUDE:
+        return !selected;
+      case NONE:
+      default:
+        return false;
+    }
+  }
+
+  /**
+   * Checks if a table is shadowed by ignoring when it is also bypassed.
+   * @param tableName the table name
+   * @param mightBeShadowedIfTrue the return value of {@link #isBypassedTable(String)}, etc
+   * @return false if the table is shadowed by ignoring, mightBeShadowedIfTrue if it is not
+   */
+  private boolean shadowedByIgnore(String tableName, boolean mightBeShadowedIfTrue) {
+    if (mightBeShadowedIfTrue && isIgnoredTable(tableName)) {
+      // todo(bowen): this warning is never triggered bc if a table is ignored,
+      // it will not be checked for bypassing, given the current implementation in Database
+      LOG.warn("Table {} is set to be bypassed but it is also ignored", tableName);
+      return false;
+    }
+    return mightBeShadowedIfTrue;
+  }
+
+  private static boolean matchedByTemplates(Set<NameTemplate> templates, String name) {
+    return templates.stream().anyMatch(p -> p.matches(name));
+  }
+
+  /**
+   * Abstract name for tables and partitions.
+   */
+  interface NameTemplate {
```

Comment:
How about `EntryName`

Reply:
Renamed in 4c0380b7c97d564304be200db2f40eb680da9f84